### ISSUE TITLE
[Bug] Doesn't work with standard functions

### ISSUE THREAD

**Issue by OwenMelbz** [spatie/once#41]
When you try to use the package outside of a class or object, it simply throws an error.

e.g.

```php
if (!function_exists('users')) {
    function users()
    {
        return once(function () {
            return User::all();
        });
    }
}
```

Will throw:

```
PHP Notice:  Undefined index: type in /Users/storm/Documents/Websites/rickard-luckin/vendor/spatie/once/src/Backtrace.php on line 44
```

If you do something like

```php
return ($this->trace['type'] ?? null) == '::';
```

To avoid the error, you then enter a rabbit hole of other errors, such as the `getObject` method etc.

**Comment by sergiz** [comment]
I wouldn't say this is a bug, but by design. Considering this was developed with Laravel usage in mind, plain functions like that are very rare.

**Comment by riasvdv** [comment]
Should be fixed in 2.2.0!
